moveit_simple/CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(moveit_simple)
+find_package(catkin REQUIRED COMPONENTS
+  roscpp
+  geometry_msgs
+  eigen_conversions
+  tf2
+  tf2_ros
+  cmake_modules
+  trajectory_msgs
+  control_msgs
+  sensor_msgs
+  moveit_core
+  moveit_ros_planning
+  moveit_visual_tools
+  actionlib
+  descartes_core
+)
+
+find_package(Boost REQUIRED)
+find_package(Eigen REQUIRED)
+
+catkin_package(
+  INCLUDE_DIRS
+    include
+  LIBRARIES
+    moveit_simple
+  CATKIN_DEPENDS
+    roscpp
+    geometry_msgs
+    eigen_conversions
+    tf2
+    tf2_ros
+    cmake_modules
+    trajectory_msgs
+    control_msgs
+    sensor_msgs
+    actionlib
+  DEPENDS
+    Boost
+    Eigen
+)
+
+include_directories(
+  include
+  ${catkin_INCLUDE_DIRS}
+  ${Boost_INCLUDE_DIRS}
+  ${Eigen_INCLUDE_DIRS}
+)
+
+add_library(moveit_simple src/moveit_simple.cpp)
+add_dependencies(moveit_simple ${catkin_EXPORTED_TARGETS})
+target_link_libraries(moveit_simple ${catkin_LIBRARIES})
+target_compile_options(moveit_simple PUBLIC -std=c++11)
```

Comment:
I'm currently living the C++11 nightmare in ros-industrial/industrial_core#156.  I think we've settled on your approach.